eng/test-configuration.json

```diff
@@ -5,6 +5,7 @@
   "retryOnRules": [
     {"testName": {"contains": "FlakyTestToEnsureRetryWorks" }},
     {"testName": {"contains": "MaxRequestBufferSizeTests.LargeUpload" }}, // tries to manipulate the state of the server's buffer
+    {"testName": {"contains": "ConnectionClosedWhenResponseDoesNotSatisfyMinimumDataRate"}}, // Issue: https://github.com/dotnet/aspnetcore/issues/58169
```

Comment:
I usually use comments to explain why a test is inherently flaky and would benefit from a retry.  If we haven't established that, I might just quarantine it.

Reply:
> If we haven't established that, I might just quarantine it.

I don't really disagree, and honestly I first created the issue to do so, but then I looked again at the operations docs and it suggested to first move it in retries and then escalate to quarantine if it doesn't help. I couldn't find anything that would suggest retrying would not help, there have been other builds without the issue even if it happened twice this week AFAIK.

Reply:
But if we decide that it's the same GH issue to track, then I agree this would also be quarantined. I will trust your judgement here since I have no other arguments to just retry.

Reply:
The other test started in the retry list and was promoted to quarantine, so I think we should probably skip straight to that.  Let's start with updating the existing bug - we can always split it out later.  Thanks!